test/revolution_test.rb

```diff
@@ -0,0 +1,522 @@
+            syskit_write task.camera_head_command_port, cmd
+        end.to do
+            have_one_new_sample task.data_out_port
+        end
+
+        expected_raw_cmd = expected_camera_head_raw_data(cmd)
+
+        json_from_sample = JSON.parse(sample.data.to_byte_array[8..-1])
+        json_from_expected = JSON.parse(expected_raw_cmd.data.to_byte_array[8..-1])
+
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["lights"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["lights"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["lasers"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["lasers"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["tilt"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["tilt"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["exposure"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["exposure"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["brightness"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["brightness"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["focus"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["focus"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["saturation"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["saturation"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["sharpness"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["sharpness"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["ratio"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["ratio"]
+        assert_equal json_from_sample["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["speed"], \
+                     json_from_expected["payload"]["devices"]["57B974C0A269"] \
+            ["cameraHead"]["camera"]["zoom"]["speed"]
```

Comment:
This is a nightmare to read, maybe transform part of the json in a variable? For example:

```
camera_from_sample = json_from_sample["payload"]["devices"]["57B...."]["cameraHead"]["camera"]
camera_from_expected = ...
assert_equal camera_from_sample["zoom"]["speed"], camera_from_expected["zoom"]["speed"]
```